File: hello-kotlin/src/main/kotlin/CliParser.kt
import java.io.File
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * Command-line argument parser
 */
object CliParser {
    fun parse(args: Array<String>): Config? {
        var inputPath: String? = null
        var outputPath: String? = null
        var serverUri = "ws://localhost:8080/audio"
        var verbose = false
        
        var i = 0
        while (i < args.size) {
            when (args[i]) {
                "--input" -> {
                    if (i + 1 < args.size) {
                        inputPath = args[++i]
                    } else {
                        Logger.error("--input requires a value")
                        return null
                    }
                }
                "--output" -> {
                    if (i + 1 < args.size) {
                        outputPath = args[++i]
                    } else {
                        Logger.error("--output requires a value")
                        return null
                    }
                }
                "--server" -> {
                    if (i + 1 < args.size) {
                        serverUri = args[++i]
                    } else {
                        Logger.error("--server requires a value")
                        return null
                    }
                }
                "--verbose" -> {
                    verbose = true
                }
                "--help" -> {
                    printHelp()
                    return null
                }
                else -> {
                    Logger.error("Unknown argument: ${args[i]}")
                    printHelp()
                    return null
                }
            }
            i++
        }
        
        if (inputPath == null) {
            Logger.error("--input is required")
            printHelp()
            return null
        }
        
        val inputFile = File(inputPath)
        if (!inputFile.exists()) {
            Logger.error("Input file does not exist: $inputPath")
            return null
        }
        
        if (!inputFile.isFile) {
            Logger.error("Input path is not a file: $inputPath")
            return null
        }
        
        if (outputPath == null) {
            val timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss"))
            val filename = inputFile.name
            outputPath = "audio/output/output-$timestamp-$filename"
        }
        
        Logger.setVerbose(verbose)
        
        return Config(
            inputPath = inputPath,
            outputPath = outputPath,
            serverUri = serverUri,
            verbose = verbose
        )
    }
    
    private fun printHelp() {
        println("""
            Audio Stream Client
            
            Usage: audio_stream_client [OPTIONS]
            
            Options:
              --input <path>      Path to input audio file (required)
              --output <path>     Path to output audio file (optional, default: audio/output/output-<timestamp>-<filename>)
              --server <uri>      WebSocket server URI (optional, default: ws://localhost:8080/audio)
              --verbose           Enable verbose logging (optional)
              --help              Display this help message
            
            Examples:
              audio_stream_client --input audio/input/test.mp3
              audio_stream_client --input audio/input/test.mp3 --output /tmp/output.mp3
              audio_stream_client --input audio/input/test.mp3 --server ws://192.168.1.100:8080/audio --verbose
        """.trimIndent())
    }
}

File: hello-php/src/CliParser.php
<?php

namespace AudioStreamClient;

class CliParser
{
    public static function parse(array $argv): ?array
    {
        $options = [
            'inputPath' => null,
            'outputPath' => null,
            'serverUri' => 'ws://localhost:8080/audio',
            'verbose' => false,
        ];

        for ($i = 1; $i < count($argv); $i++) {
            switch ($argv[$i]) {
                case '--input':
                    if ($i + 1 < count($argv)) {
                        $options['inputPath'] = $argv[++$i];
                    } else {
                        Logger::error('--input requires a value');
                        return null;
                    }
                    break;
                case '--output':
                    if ($i + 1 < count($argv)) {
                        $options['outputPath'] = $argv[++$i];
                    } else {
                        Logger::error('--output requires a value');
                        return null;
                    }
                    break;
                case '--server':
                    if ($i + 1 < count($argv)) {
                        $options['serverUri'] = $argv[++$i];
                    } else {
                        Logger::error('--server requires a value');
                        return null;
                    }
                    break;
                case '--verbose':
                case '-v':
                    $options['verbose'] = true;
                    break;
                case '--help':
                case '-h':
                    self::printHelp();
                    return null;
                default:
                    Logger::error('Unknown argument: ' . $argv[$i]);
                    self::printHelp();
                    return null;
            }
        }

        if ($options['inputPath'] === null) {
            Logger::error('--input is required');
            self::printHelp();
            return null;
        }

        if (!file_exists($options['inputPath'])) {
            Logger::error('Input file does not exist: ' . $options['inputPath']);
            return null;
        }

        if ($options['outputPath'] === null) {
            $timestamp = date('Ymd-His');
            $filename = basename($options['inputPath']);
            $options['outputPath'] = "audio/output/output-$timestamp-$filename";
        }

        return $options;
    }

    private static function printHelp(): void
    {
        echo <<<HELP
Audio Stream Client

Usage: php audio_stream_client.php [OPTIONS]

Options:
  --input <path>      Path to input audio file (required)
  --output <path>     Path to output audio file (optional, default: audio/output/output-<timestamp>-<filename>)
  --server <uri>      WebSocket server URI (optional, default: ws://localhost:8080/audio)
  --verbose, -v       Enable verbose logging (optional)
  --help, -h          Display this help message

Examples:
  php audio_stream_client.php --input audio/input/test.mp3
  php audio_stream_client.php --input audio/input/test.mp3 --output /tmp/output.mp3
  php audio_stream_client.php --input audio/input/test.mp3 --server ws://192.168.1.100:8080/audio --verbose

HELP;
    }
}
